src/main.rs

```diff
@@ -31,8 +33,38 @@ use vmm::vmm_config::instance_info::{InstanceInfo, InstanceState};
 
 const DEFAULT_API_SOCK_PATH: &str = "/tmp/firecracker.socket";
 const DEFAULT_INSTANCE_ID: &str = "anonymous-instance";
+const KERNEL_VERSION: f64 = 4.14;
+
+fn is_kvm_writable() -> Result<bool, Error> {
+    let kvm_rw = Command::new("sh")
+        .arg("-c")
+        .arg("[ -r /dev/kvm ] && [ -w /dev/kvm ] && echo OK|| echo FAIL")
```

Comment:
This is on the Firecracker critical fast-path so it should take speed very seriously.
Starting outside commands and parsing their output is slow.
File permissions can be retrieved using https://doc.rust-lang.org/std/os/unix/fs/trait.PermissionsExt.html#tymethod.mode